### PR TITLE
Addressed comments on s8 task observed during testing

### PR DESCRIPTION
**Pull request by rsarwad**
"[lte][agw]: Addressed comments on s8 task observed during testing

## Summary
1. Added ULI and serving network in deletion session request
2. Added pco in create session request and processing pco which is received in create session response
3. Removed explicit length and num_protocol_container_id field in pco
4. Fixed the memory leak found during testing
## Test Plan
Executed s1ap sanity test suite

### CODE REVIEW

File: lte/gateway/c/oai/lib/s8_proxy/s8_client_api.cpp [hunk 1]
```diff
@@ -455,12 +501,17 @@ static void convert_proto_msg_to_itti_csr(
   } else {
     s5_response->cause = REQUEST_ACCEPTED;
   }
+  if (response.has_protocol_configuration_options()) {
+    get_pco_from_proto_msg(
+        response.protocol_configuration_options(), &s5_response->pco);
+  }
```
Comment (uri200) [inline lte/gateway/c/oai/lib/s8_proxy/s8_client_api.cpp:502]: I can't comment the line 499, but shoudn't we be checking `if (response.has_gtp_error())` as the very beginning of this function? 

If we have gtp_error, the other field are going to be bad. Only gtp_error is good. So no need to go through the whole function.

File: lte/gateway/c/oai/tasks/mme_app/mme_app_itti_messaging.c [hunk 2]
```diff
@@ -361,19 +361,8 @@ int mme_app_send_s11_create_session_req(
...
-  session_request_p->serving_network.mcc[2] =
-      ue_mm_context->e_utran_cgi.plmn.mcc_digit3;
-  session_request_p->serving_network.mnc[0] =
-      ue_mm_context->e_utran_cgi.plmn.mnc_digit1;
-  session_request_p->serving_network.mnc[1] =
-      ue_mm_context->e_utran_cgi.plmn.mnc_digit2;
-  session_request_p->serving_network.mnc[2] =
-      ue_mm_context->e_utran_cgi.plmn.mnc_digit3;
+  COPY_PLMN_IN_ARRAY_FMT(
+      (session_request_p->serving_network), (ue_mm_context->e_utran_cgi.plmn));
```
Comment (uri200) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_itti_messaging.c:365]: 👍

**Review by uri200 — CHANGES_REQUESTED** [review]

File: lte/gateway/c/oai/lib/s8_proxy/s8_client_api.cpp [hunk 3]
```diff
@@ -195,6 +195,30 @@ static void recv_s8_create_session_response(
...
+  mcc[0] = convert_digit_to_char(itti_msg_sn.mcc[0]);
+  mcc[1] = convert_digit_to_char(itti_msg_sn.mcc[1]);
+  mcc[2] = convert_digit_to_char(itti_msg_sn.mcc[2]);
+  mnc[0] = convert_digit_to_char(itti_msg_sn.mnc[0]);
+  mnc[1] = convert_digit_to_char(itti_msg_sn.mnc[1]);
+  if ((itti_msg_sn.mnc[2] & 0xf) != 0xf) {
+    mnc[2]  = convert_digit_to_char(itti_msg_sn.mnc[2]);
+    mnc_len = 3;
+  } else {
+    mnc[2]  = '\0';
```
Comment (ulaskozat) [inline lte/gateway/c/oai/lib/s8_proxy/s8_client_api.cpp]: looks already initialized to null (ascii 0) on line 202.

**Review by ulaskozat — APPROVED** [review]
LGTM

**Review by uri200 — APPROVED** [review]

**Pull request merged**